From aaed6c3668bf439997731040e39ba305b9284e32 Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Tue, 24 Dec 2024 11:36:31 +0800
Subject: [PATCH 063/104] mtk: mt76: mt7996: do not check enable_beacon when
 adding BSS_INFO

We use info->enable_beacon to control the timing of beacon TX. However,
adding/updating BSS_INFO to FW do not affect the beacon TX, so it is not
necessary to check the enable_beacon before mt7996_mcu_add_bss_info().

Wihtout the commit, the update about the BSS before beacon enabled
is not sent to FW.

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 mt7996/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mt7996/main.c b/mt7996/main.c
index bc068ef7..26b930d6 100644
--- a/mt7996/main.c
+++ b/mt7996/main.c
@@ -1016,7 +1016,7 @@ static void mt7996_link_info_changed(struct ieee80211_hw *hw,
 	 * and then peer references bss_info_rfch to set bandwidth cap.
 	 */
 	if ((changed & BSS_CHANGED_BSSID && !is_zero_ether_addr(info->bssid)) ||
-	    (changed & BSS_CHANGED_BEACON_ENABLED && info->enable_beacon)) {
+	    (changed & BSS_CHANGED_BEACON_ENABLED)) {
 		mt7996_mcu_add_bss_info(phy, vif, info, mconf, mlink, true);
 		mt7996_mcu_add_sta(dev, vif, info, mconf, NULL, mlink,
 				   CONN_STATE_PORT_SECURE,
-- 
2.45.2

